nixos/modules/services/networking/seafile.nix

```diff
@@ -0,0 +1,272 @@
+        This is appended under the fileserver section.
+        Refer to <link xlink:href="https://manual.seafile.com/config/seafile-conf/" />
+        for all available options.
+      '';
+    };
+
+    adminEmail = mkOption {
+      example = "john@example.com";
+      type = types.str;
+      description = ''
+        Seafile Seahub Admin Account Email.
+      '';
+    };
+
+    initialAdminPassword = mkOption {
+      example = "someStrongPass";
+      type = types.str;
+      description = ''
+        Seafile Seahub Admin Account initial password.
+        Should be change via Seahub web front-end.
+      '';
+    };
+
+    seahubSecretKeyFile = mkOption {
+      type = types.path;
+      description = ''
+        A file containing the django secret key for safe session.
+      '';
+    };
+
+    seahubExtraConf = mkOption {
+      default = "";
+      type = types.lines;
+      description = ''
+        Extra config to append to `seahub_settings.py` file.
+        Refer to <link xlink:href="https://manual.seafile.com/config/seahub_settings_py/" />
+        for all available options.
+      '';
+    };
+  };
+
+  ###### Implementation
+
+  config = mkIf cfg.enable {
+
+    systemd.targets.seafile = {
+      wantedBy = [ "multi-user.target" ];
+      description = "Seafile components";
+    };
+
+    systemd.services.seaf-server = {
+      description = "Seafile server";
+      partOf = [ "seafile.target" ];
+      after = [ "network.target" ];
+      wantedBy = [ "seafile.target" ];
+      serviceConfig = {
+        User = "seafile";
+        Group = "seafile";
+        DynamicUser = true;
+        ProtectHome = true;
+        PrivateUsers = true;
+        PrivateDevices = true;
+        ProtectClock = true;
+        ProtectHostname = true;
+        ProtectProc = "invisible";
+        ProtectKernelModules = true;
+        ProtectKernelTunables = true;
+        ProtectKernelLogs = true;
+        ProtectControlGroups = true;
+        RestrictNamespaces = true;
+        RestrictAddressFamilies = [ "AF_UNIX" "AF_INET" ];
+        LockPersonality = true;
+        RestrictRealtime = true;
+        RestrictSUIDSGID = true;
+        MemoryDenyWriteExecute = true;
+        SystemCallArchitectures = "native";
+        StateDirectory = "seafile";
+        RuntimeDirectory = "seafile";
+        LogsDirectory = "seafile";
+        Type = "forking";
+        ExecStart = ''
+          ${pkgs.seafile-server}/bin/seaf-server \
+          -F ${centralConfigDir} \
+          -c ${ccnetDir} \
+          -d ${dataDir} \
+          -l /var/log/seafile/server.log \
+          -P /run/seafile/server.pid \
+          -p /run/seafile
+        '';
+        PIDFile = "/run/seafile/server.pid";
+      };
+      preStart = ''
+        mkdir -p ${seafRoot}/{conf,data}
+        mkdir -p ${dataDir}/library-template
+        ln -sf ${ccnetConf} ${centralConfigDir}/ccnet.conf
+        ln -sf ${seafileConf} ${centralConfigDir}/seafile.conf
```

Comment:
Is there any possible way to provide the location of configuration files to the programs which need them?